source/slang/slang-ir-spirv-legalize.cpp

```diff
@@ -953,13 +957,26 @@ struct SPIRVLegalizationContext : public SourceEmitterBase
 
             // If we reach here, we need to allocate a temp var.
             auto tempVar = builder.emitVar(ptrType->getValueType());
+            builder.addDecoration(tempVar, kIROp_DisableCopyEliminationDecoration);
```

Comment:
I would definitely like to see a comment here explaining why adding this decoration is important.

Reply:
The whole purpose of the logic here is to introduce this temp var for spirv to be valid. If we allow later optimziation pass to remove the var, then that makes this step pointless.